### PR TITLE
Update tooling and and add nix

### PR DESCRIPTION
**Pull request by stoeffel**
adds nix, make and gh-action to run tests:
To develop use: `nix develop --command make watch`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,33 @@
+npm-install:
```
Comment (gampleman) [inline Makefile:1]: Generally speaking it's a good idea to declare these as `.PHONY` in a makefile, you can skip errors when a file called `test` is accidentally created your tests won't run.

File: bin/cli.js [hunk 2]
```diff
@@ -25,15 +25,16 @@ var argv = require("yargs")
...
     if (typeof arg === "string") return arg.split(" ");
     return [];
   })
   .default("elm-test-args", []).argv;
 
 var model = runner.init(argv);
 
-runner.run(model, function(warnings) {
+runner.run(model, function (warnings) {
+  console.log("warnings", warnings);
```
Comment (gampleman) [inline bin/cli.js:37]:
```suggestion
  console.warn("warnings", warnings);
```

AFAIK it goes to stderr in node, which is probably more desirable.

File: bin/runner.js [hunk 3]
```diff
@@ -188,32 +188,35 @@ function writeFile(testsDocPath) {
 
     var testsDocModulePath = path.join(testsDocPath, modulePath.join("/"));
 
-    mkdirp(testsDocModulePath, function(err) {
-      if (err) {
+    mkdirp(testsDocModulePath)
+      .then(function () {
+        fs.writeFile(
```
Comment (gampleman) [inline bin/runner.js:193]: Might be easier to eventually rewrite this in async/await style, but that's perhaps for another PR...

**Review by gampleman — APPROVED** [review]

File: bin/runner.js (hunk 3 above)
Comment (stoeffel) [reply bin/runner.js:193]: yes!

**Pull request merged**